Frameworks/CoreGraphics/CGContext.mm

```diff
@@ -1015,13 +1019,132 @@ void CGContextClipToRects(CGContextRef context, const CGRect* rects, unsigned co
     CGContextClip(context);
 }
 
+HRESULT __CGContext::ClipToD2DMaskBitmap(ID2D1Bitmap* bitmap, CGRect rect, bool shouldInterpolate) {
+    RETURN_HR_IF_NULL(E_INVALIDARG, bitmap);
+
+    D2D1_SIZE_U bitmapSize = bitmap->GetPixelSize();
+
+    RETURN_HR_IF(E_INVALIDARG, bitmapSize.width == 0 || bitmapSize.height == 0);
+
+    auto& state = CurrentGState();
```

Comment:
I think this would damage clarity: it could be `__CGContextDrawingState&` or `__CGContext::__CGContextDrawingState&`. Having it on the right hand side is clearer (in my opinion)